### ISSUE TITLE
RuntimeError: Tensors must have same number of dimensions: got 3 and 4

### ISSUE THREAD

**Issue by theUpsider** [oobabooga/text-generation-webui#288]
I get an exception as soon as I try to generate something. This is a linux server with python 3.10.9.

`python server.py --chat --listen --gpu-memory 45 45 45 45
Loading the extension "gallery"... Ok.
The following models are available:

1. llama-30b-hf
2. llama-65b-hf

Which one do you want to load? 1-2

1

Loading llama-30b-hf...
Loading checkpoint shards: 100%|████████████████████████████████████| 61/61 [04:04<00:00,  4.02s/it]
Loaded the model in 245.67 seconds.
Running on local URL:  http://0.0.0.0:7860

/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/transformers/generation/utils.py:1201: UserWarning: You have modified the pretrained model configuration to control generation. This is a deprecated strategy to control generation and will be removed soon, in a future version. Please use a generation configuration file (see https://huggingface.co/docs/transformers/main_classes/text_generation)
  warnings.warn(
Exception in thread Thread-4 (gentask):
Traceback (most recent call last):
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/mnt/media/models/one-click-installers-oobabooga-linux/text-generation-webui/modules/callbacks.py", line 64, in gentask
    ret = self.mfunc(callback=_callback, **self.kwargs)
  File "/mnt/media/models/one-click-installers-oobabooga-linux/text-generation-webui/modules/text_generation.py", line 191, in generate_with_callback
    shared.model.generate(**kwargs)
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/transformers/generation/utils.py", line 1452, in generate
    return self.sample(
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/transformers/generation/utils.py", line 2468, in sample
    outputs = self(
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/accelerate/hooks.py", line 165, in new_forward
    output = old_forward(*args, **kwargs)
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 772, in forward
    outputs = self.model(
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 621, in forward
    layer_outputs = decoder_layer(
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/accelerate/hooks.py", line 165, in new_forward
    output = old_forward(*args, **kwargs)
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 318, in forward
    hidden_states, self_attn_weights, present_key_value = self.self_attn(
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/accelerate/hooks.py", line 165, in new_forward
    output = old_forward(*args, **kwargs)
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 228, in forward
    query_states, key_states = apply_rotary_pos_emb(query_states, key_states, cos, sin, offset=offset)
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 142, in apply_rotary_pos_emb
    q_embed = (q * cos) + (rotate_half(q) * sin)
  File "/home/fischerd/anaconda3/envs/textgen3/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 136, in rotate_half
    return torch.cat((-x2, x1), dim=-1)
RuntimeError: Tensors must have same number of dimensions: got 3 and 4`

- I tried installing the custom push for llama.
- I had to install `conda install protobuf`
- Google module was not found so I used the install script from the automated linux installer.
- I tried a fresh install only using the linux installer

**Comment by dnhkng** [comment]
Found the fix in the issue #282:
```
pip uninstall transformers 
pip install git+https://github.com/zphang/transformers.git@68d640f7c368bcaaaecfc678f11908ebbd3d6176
```

**Comment by MolotovCherry** [comment]
> Found the fix in the issue #282:
> 
> ```
> pip uninstall transformers 
> pip install git+https://github.com/zphang/transformers.git@68d640f7c368bcaaaecfc678f11908ebbd3d6176
> ```

llama wouldn't work for me. This was the solution. Thanks a lot!

**Comment by enn-nafnlaus** [comment]
> Found the fix in the issue #282:
> 
> ```
> pip uninstall transformers 
> pip install git+https://github.com/zphang/transformers.git@68d640f7c368bcaaaecfc678f11908ebbd3d6176
> ```

Yeay!  Now I'm off to a different error!  :)
